website/docs/getting-started/deploy-mesh-gateway.mdx

```diff
@@ -0,0 +1,184 @@
+---
+id: deploy-mesh-gateway
+title: "How to: Deploy a GraphQL Mesh Gateway"
+sidebar_label: Deploy a Mesh Gateway
+---
+
+Thanks to its flexible architecture and embedded server relying on [GraphQL Yoga](https://www.graphql-yoga.com/) and [Envelop](https://www.envelop.dev/), **GraphQL Mesh can be deployed anywhere**!
+
+We already saw that `mesh dev` can be used for local development.
+
+Simirarly, Mesh provides a `mesh start` CLI command for production environments.
+
+`mesh start` can be used for all environments that supports starting a webserver (Heroku, Digital Ocean, etc).
+
+Setup Mesh on a Serverless environment requires some integration work, detailled below.
+
+
+<p>&nbsp;</p>
+
+------
+
+<p>&nbsp;</p>
+
+
+## Deploy Mesh with `mesh start`
+
+While `mesh dev` handles the generation of the SDK code, `mesh start` expect to load the Gateway schema and runtime from the `.mesh/` folder.
+
+This mechanism helps:
+- __reducing the start time of the server__: no build step is required
+- __preventing starting failure if one of the Source is unreachable__: we don't fetch the API definition file at startup, ensuring that the fetched definitions are validated at build time
+
+
+To deploy a Mesh Gateway, you need to ensure that `mesh build` is called during the deployment, for example with a `prebuild` step:
+
+_package.json_
+```json
+{
+  "scripts": {
+    "start": "mesh start",
+    // will be run during deployment
+    "prebuild": "mesh build"
+  }
+}
+```
+
+<p>&nbsp;</p>
+
+------
+
+<p>&nbsp;</p>
+
+
+## Deploy Mesh on Serverless
+
+Serverless deployment require some integration since we cannot keep the `mesh start` server running.
+
+
+<p>&nbsp;</p>
+
+### Deploy Mesh on Vercel with Next.js API Routes
+
+First, let's ensure that `mesh build` will be run during deployment.
+
+Vercel - list most platforms, run `yarn build` for deployment.
+For this reason, we will hook to it with the `prebuild` step:
+
+_package.json_
+```json
+{
+  "scripts": {
+    "start": "mesh start",
+    // will be run during deployment
+    "prebuild": "mesh build"
+  }
+}
+```
+
+<br />
+
+Now, let's integrate our Mesh Gateway in a Next.js API Routes using GraphQL Yoga:
+
+_website/src/pages/api/graphql/index.ts_
+```ts
+import { createServer } from '@graphql-yoga/node';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getBuiltMesh } from './.mesh';
+
+
+
+export default async function apiHandler(req: NextApiRequest, res: NextApiResponse) {
+  // retrieve the mesh instance (with configured Envelop plugins)
+  const mesh = await getBuiltMesh();
```

Comment:
This would cause Mesh run from scratch everytime that function is called.

Reply:
Good catch! I was wondering why you put it outside, now I know 😅